File: frontend/src/lib/types/briefing.ts
export type BriefingSummary = {
  id: number;
  date: string;
  model: string;
};

export type Briefing = BriefingSummary & {
  session_id: string | null;
  content_markdown: string;
};

export type GenerateBriefingResult = {
  id: number;
  session_id: string;
  content_markdown: string;
  model: string;
  date: string;
  tool_calls: Array<{ name: string; input: Record<string, unknown> }>;
};

File: backend/app/schemas/briefing.py
"""Briefing schemas."""

from datetime import datetime

from pydantic import BaseModel


class BriefingOut(BaseModel):
    id: int
    date: datetime
    session_id: str | None
    content_markdown: str
    model: str

    class Config:
        from_attributes = True


class BriefingSummary(BaseModel):
    """Lightweight list entry (no full content)."""
    id: int
    date: datetime
    model: str

    class Config:
        from_attributes = True


class GenerateBriefingResult(BaseModel):
    id: int
    session_id: str
    content_markdown: str
    model: str
    date: str
    tool_calls: list[dict] = []
